File: app/src/main/java/closer/vlllage/com/closer/CloserFirebaseInstanceIDService.java
package closer.vlllage.com.closer;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

import closer.vlllage.com.closer.handler.data.AccountHandler;

public class CloserFirebaseInstanceIDService extends FirebaseInstanceIdService {
    @Override
    public void onTokenRefresh() {
        String deviceToken = FirebaseInstanceId.getInstance().getToken();
        ((App) getApplication()).$(AccountHandler.class).updateDeviceToken(deviceToken);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/CloserFirebaseInstanceIDService.kt
package closer.vlllage.com.closer

import closer.vlllage.com.closer.handler.data.AccountHandler
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService

class CloserFirebaseInstanceIDService : FirebaseInstanceIdService() {
    override fun onTokenRefresh() {
        val deviceToken = FirebaseInstanceId.getInstance().token
        (application as App).`$`(AccountHandler::class.java).updateDeviceToken(deviceToken!!)
    }
}
